sphinx/ext/duration.py

```diff
@@ -79,18 +89,50 @@ def on_doctree_read(app: Sphinx, doctree: nodes.document) -> None:
 def on_build_finished(app: Sphinx, error: Exception) -> None:
     """Display duration ranking on the current build."""
     domain = app.env.domains['duration']
-    if not domain.reading_durations:
+    reading_durations = domain.reading_durations
+    if not reading_durations:
         return
-    durations = sorted(
-        domain.reading_durations.items(), key=itemgetter(1), reverse=True
-    )
 
-    logger.info('')
-    logger.info(
-        __('====================== slowest reading durations =======================')
-    )
-    for docname, d in islice(durations, 5):
-        logger.info(f'{d:.3f} {docname}')  # NoQA: G004
+    # Get default options and update with user-specified values
+    options = DEFAULT_OPTIONS.copy()
+    user_options = getattr(app.config, 'duration_options', DEFAULT_OPTIONS)
+    options.update(user_options)
+
+    if options['print_total']:
+        logger.info('')
+        logger.info(
+            __(
+                '====================== total reading duration =========================='
+            )
+        )
+
+        n_files = len(reading_durations)
+        s = '' if n_files == 1 else 's'
+
+        total = sum(reading_durations.values())
+        logger.info('Total time reading %d file%s:\n', n_files, s)
+        logger.info(_format_seconds(total, multiline=True))
+
+    if options['print_slowest']:
+        logger.info('')
+        logger.info(
+            __(
+                '====================== slowest reading durations ======================='
+            )
+        )
+        sorted_durations = sorted(
+            reading_durations.items(), key=itemgetter(1), reverse=True
+        )
+        n_slowest = options['n_slowest']
+        n_slowest = len(sorted_durations) if n_slowest == -1 else n_slowest
+        for docname, d in islice(sorted_durations, n_slowest):
+            logger.info('%s %s', _format_seconds(d), docname)
+
+    if options['write_durations']:
+        # Write to JSON
+        out_file = Path(app.builder.outdir) / 'sphinx_durations.json'
```

Comment:
Should we save to source dir instead? Or maybe add an option to choose?
```suggestion
        out_file = Path(app.builder.srcdir) / 'sphinx_durations.json'
```